File: OOP/7_Function_Overriding.cpp



#include<bits/stdc++.h>
using namespace std;

// parent class
class A{
    public:
    string name;
    int age;

    //constructor
    A(string name, int age){
        this->name = name;
        this->age = age;
    }
    void display(){
        cout<<"Your is name"<< name <<" and age " <<age<<endl;
    }

};

//child class
class B : public A{
  public:
    string designation;
  // constructor
    B(string name, int age,string designation) : A(name, age) {
      this->designation=designation;
    }
 // display method override from  parent class
    void display(){
        cout<<"Your is name"<< name <<" and age "<<age <<" Designation "<< designation<<endl;
    }
};


int main(){
  
  B obj("John", 25,"Student");
  obj.display();
    return 0;
}

File: OOP/7_Function_Overriding.py

class A:
  def __init__(self,name,age) -> None:
    self.name=name
    self.age=age
  
  def display(self):
    print(f"{self.name} {self.age}")


class B(A):
  def __init__(self, name, age,designation) -> None:
    super().__init__(name, age)
    self.designation=designation

  # display method override from class A
  def display(self): 
    print(f"{self.name} {self.age} {self.designation}")



person =B('rafiul',32,'student')
person.display()




# Runtime Polymorphism -> Dynamic Binding | Overriding | Late Binding

# CompileTime Polymorphism-> Static Binding | Overloading | Early Binding